### PR TITLE
#Network level hints 2: Cleanup: Netloc utilities and network level comm_split_type hints 

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**
1. Added asserts after some malloc calls.
2. Removed an unused argument in utility function.

### CODE REVIEW

File: src/mpi/init/netloc_util.c [hunk 1]
```diff
@@ -402,6 +402,81 @@ int MPIR_Netloc_get_network_end_point(MPIR_Netloc_network_attributes network_att
...
+{
+    int mpi_errno = MPI_SUCCESS;
+    int max_level = 0;
+    netloc_node_t **traversal_order;
+    int start_index, end_index, i, j, node_index, num_nodes;
+
+    traversal_order =
+        (netloc_node_t **) MPL_malloc(sizeof(netloc_node_t *) * topology->num_nodes, MPL_MEM_OTHER);
+
+    MPIR_Assert(traversal_order != NULL);
```
Comment (raffenet) [inline src/mpi/init/netloc_util.c]: Asserting is kind of heavy-handed. We have some macros like `MPIR_CHKPMEM_MALLOC` that can help with allocations and return useful error codes. MPICH will still crash in the default case, of course, but we should still attempt to return errors when possible.

Comment (KavithaTipturMadhu) [reply src/mpi/init/netloc_util.c]: Used MPIR_CHKPMEM_MALLOC in place of MPL_malloc and assert

**Comment by KavithaTipturMadhu** [comment]
@raffenet Can you take a look at this?

**Comment by KavithaTipturMadhu** [comment]
test:jenkins/ch4/all

**Pull request merged**